master/internal/rm/kubernetesrm/resource_pool_intg_test.go

```diff
@@ -0,0 +1,238 @@
+			require.Equal(t, int(sproto.SchedulingStateQueued), rp.allocationIDToRunningPods[newAllocID])
+			require.Zero(t, rp.allocationIDToRunningPods[newAllocID])
+
+			req := sproto.UpdatePodStatus{
+				ContainerID: "bogus",
+				State:       tt.state,
+			}
+			// TODO (bradley, pods2jobs): new implementation won't require container ID
+			req.ContainerID = string(rp.allocationIDToContainerID[newAllocID])
+
+			rp.UpdatePodStatus(req)
+
+			// If scheduled (backfilled or scheduled), check that running pods++
+			require.Equal(t, tt.runningPod, rp.allocationIDToRunningPods[newAllocID])
+		})
+	}
+}
+
+func TestPendingPreemption(t *testing.T) {
+	rp, jobID := testResourcePoolWithJob(t, defaultSlots)
+	allocID := newTestPod(t, rp, jobID)
+	err := rp.PendingPreemption(sproto.PendingPreemption{
+		AllocationID: allocID,
+	})
+	require.Equal(t, rmerrors.ErrNotSupported, err)
+}
+
+func TestSetGroupWeight(t *testing.T) {
+	rp, _ := testResourcePoolWithJob(t, defaultSlots)
+	err := rp.SetGroupWeight(sproto.SetGroupWeight{})
+	require.Equal(t, rmerrors.UnsupportedError("set group weight is unsupported in k8s"), err)
+}
+
+func TestSetGroupPriority(t *testing.T) {
+	rp, _ := testResourcePoolWithJob(t, defaultSlots)
+
+	cases := []struct {
+		name        string
+		newPriority int
+		preemptible bool
+	}{
+		{"not-preemptible", 0, false},
+		{"no change", int(config.KubernetesDefaultPriority), true},
+		{"increase", 100, true},
+		{"decrease", 1, true},
+		{"negative", -10, true}, // doesn't make sense, but it is allowed
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			jobID := model.NewJobID()
+
+			rp.AllocateRequest(sproto.AllocateRequest{
+				JobID:       jobID,
+				Preemptible: tt.preemptible,
+			})
+
+			err := rp.SetGroupPriority(sproto.SetGroupPriority{
+				Priority:     tt.newPriority,
+				ResourcePool: rp.poolConfig.PoolName,
+				JobID:        jobID,
+			})
+
+			if tt.preemptible {
+				require.NoError(t, err)
+				// TODO (bradley): check that the priority change is reflected in rm events
+				// require.Equal(t, tt.newPriority, *rp.getOrCreateGroup(jobID).Priority)
+			} else {
+				require.Error(t, err)
+			}
+		})
+	}
+}
+
+func TestValidateResources(t *testing.T) {
+	cases := []struct {
+		name        string
+		slots       int
+		fulfillable bool
+	}{
+		{"valid", 1, true},
+		{"invalid", 100, false},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			rp, _ := testResourcePoolWithJob(t, defaultSlots)
+			res := rp.ValidateResources(sproto.ValidateResourcesRequest{
+				Slots: tt.slots,
+			})
+			require.Equal(t, tt.fulfillable, res.Fulfillable)
+		})
+	}
+}
+
+func TestSchedule(t *testing.T) {
+	rp, jobID := testResourcePoolWithJob(t, defaultSlots)
+
+	allocID := newTestPod(t, rp, jobID)
+	require.True(t, rp.reschedule)
+	require.False(t, rp.reqList.IsScheduled(allocID))
+
+	rp.Schedule()
+
+	require.False(t, rp.reschedule)
+	go require.True(t, rp.reqList.IsScheduled(allocID))
```

Comment:
Why is this one a goroutine and not the others? (I apologize, I'm not familiar with this part of the code. I dug into it a bit, but couldn't find anything.)

Reply:
I think @carolinaecalderon was trying to fix the race condition. This (surprisingly) tricks the race detector but, of course, doesn't fix it. I think she's working on something else.

Reply:
yeah, outdated -- I removed the test that was flaky in CI. I created a new ticket (https://hpe-aiatscale.atlassian.net/browse/RM-291) where I'll address this later.